discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -91,7 +96,40 @@ impl SqliteStorage {
         Ok(Self { pool })
     }
 
-    async fn begin_tx(&self) -> Result<Transaction<'_, Sqlite>, Error> {
+    async fn setup_keep_stacks_in_sync(tx: &mut Transaction<'_, Sqlite>) -> Result<(), Error> {
+        let expected_ids = &[
+            stack::ops::breaking::BreakingNews::id(),
+            stack::ops::personalized::PersonalizedNews::id(),
+            stack::ops::trusted::TrustedNews::id(),
+            stack::exploration::Stack::id(),
+        ];
+
+        let mut query_builder = QueryBuilder::new(String::new());
+        query_builder
+            .push("INSERT INTO Stack (stackId) ")
+            .push_values(expected_ids, |mut stm, id| {
+                stm.push_bind(id);
+            })
+            .push("ON CONFLICT DO NOTHING")
+            .build()
+            .persistent(false)
+            .execute(&mut *tx)
+            .await
+            .map_err(|err| Error::Database(err.into()))?;
+
+        query_builder
+            .reset()
+            .push("DELETE FROM Stack WHERE stackId not in ")
```

Comment:
❓ shouldn't key words be upper case?
```suggestion
            .push("DELETE FROM Stack WHERE stackId NOT IN ")
```

Reply:
:rocket:  yes, stylistically that would be better